tests/api/test_cluster_descriptor.cpp

```diff
@@ -43,6 +46,22 @@ std::unique_ptr<tt_ClusterDescriptor> get_cluster_desc() {
     return cluster_desc;
 }
 
+TEST(ApiTest, DetectArch) {
+    // TODO: This should be part of cluster descriptor. It is currently used like this from tt_metal.
+    tt::ARCH arch = detect_arch();
+
+    // Expect it to be invalid if no devices are found.
+    if (PCIDevice::enumerate_devices().empty()) {
+        EXPECT_EQ(arch, tt::ARCH::Invalid);
+    } else {
+        EXPECT_NE(arch, tt::ARCH::Invalid);
+
+        // TODO: This should be the only available API, previous call should be routed to this one to get any arch.
+        tt::ARCH arch2 = detect_arch(PCIDevice::enumerate_devices()[0]);
+        EXPECT_NE(arch2, tt::ARCH::Invalid);
```

Comment:
If I am reading the code correctly, we can also add line 
`EXPECT_EQ(arch, arch2)`

Reply:
While that will be true in our current setup/code, that is wrong from purely the API perspective. 
The detect_arch() returns ANY arch found in the cluster, and with another call I specified a specific PCI device for which I want arch for.
Which is the reason to avoid the first call (without arguments) altogether probably.

Reply:
> Which is the reason to avoid the first call (without arguments) altogether probably.

I think this is the main point. But I still think it would still be nice to check that `detect_arch()` returns at least one of the arch that you get when you iterate ove PCI devices

Reply:
I've added the initially suggested check. This will get removed anyway, so hence my indifference.